kedro/template/{{ cookiecutter.repo_name }}/kedro_cli.py

```diff
@@ -122,7 +122,19 @@ def _split_string(ctx, param, value):
     return [item for item in value.split(",") if item]
 
 
+def _convert_text_to_numeric(value):
```

Comment:
Like that you've factored this out, but not a big fan of the name here, sounds like it expects the value to be a numeric-parseable string rather than only converting to a numeric type _if_ possible and keeping it as a string otherwise. That said, I don't have a better suggestion, so feel free to ignore me...

Reply:
I think that is actually a good point. Maybe `_convert_str_type_to_numeric(value):` or even `_try_convert_str_type_to_numeric(value):`?

Reply:
On second thought, maybe just keep the name and add a docstring?

Reply:
`_try_convert_to_numeric`? It technically also works for inputs that are not strings.

Reply:
will go with @DmitriiDeriabinQB last suggestion.